mmhuman3d/data/data_structures/smc_reader.py

```diff
@@ -222,22 +245,58 @@ def get_iphone_extrinsics(self, iphone_id=0, homogeneous=True):
                     keys are R and T,
                     each value is an ndarray.
         """
-        assert iphone_id == 0, 'Currently only one iPhone.'
+        if iphone_id != 0:
+            raise KeyError('Currently only one iPhone.')
         R = np.asarray(self.calibration_dict['iPhone']['R']).reshape(3, 3)
         T = np.asarray(self.calibration_dict['iPhone']['T']).reshape(3)
 
+        # cam2world
+        extrinsics = np.identity(4, dtype=float)
+        extrinsics[:3, :3] = R
+        extrinsics[:3, 3] = T
+
+        if vertical:
+            # 90-degree clockwise rotation around z-axis
+            R = np.eye(4)
+            R[:2, :2] = np.array([[0, -1], [1, 0]])
+            # Note the extrinsics is cam2world
+            extrinsics = extrinsics @ np.linalg.inv(R)
```

Comment:
Have we tested this extrinsic?

Reply:
It is included in the [`test_iphone_rotation()` ](https://github.com/open-mmlab/mmhuman3d/blob/feature_load_smc/tests/test_smc_reader.py#L428-L431).